security/backends/elasticsearch/writer.py

```diff
@@ -432,23 +446,43 @@ def clean_logs(self, type, timestamp, backup_path, stdout):
-                        if storage.exists('{}.json.gz'.format(log_file_name)):
-                            i = 1
-                            while storage.exists('{}({}).json.gz'.format(log_file_name, i)):
-                                i += 1
-                            log_file_name = '{}({})'.format(log_file_name, i)
-                        stdout.write(4 * ' ' + 'generating backup file: {}.json.gz'.format(log_file_name))
-                        with storage.open('{}.json.gz'.format(log_file_name), 'wb') as f:
-                            with TextIOWrapper(gzip.GzipFile(filename='{}.json'.format(log_file_name),
-                                                             fileobj=f, mode='wb')) as gzf:
-                                json.dump(
-                                    lazy_serialize_iterable(batch_data), gzf, cls=DjangoJSONEncoder, indent=5
-                                )
-                stdout.write(4 * ' ' + 'deleting logs')
-                batch_delete_queryset(qs_filtered_by_day, settings.ELASTICSERACH_DELETE_BATCH_SIZE)
+                    created_log_files = []
+
+                    try:
+                        for batch_data in batch(
+                                get_queryset_dict_data_by_batch(
+                                    qs_filtered_by_day, settings.ELASTICSERACH_CLEAN_LOGS_READ_BATCH_SIZE
+                                ),
+                                settings.CLEAN_LOGS_BACKUP_FILE_BATCH_SIZE):
+                            log_file_name = os.path.join(backup_path, str(step_timestamp.date()))
+                            if storage.exists(f'{log_file_name}.json.gz'):
+                                i = 1
+                                while storage.exists(f'{log_file_name}({i}).json.gz'):
+                                    i += 1
+                                log_file_name = f'{log_file_name}({i})'
+                            log_file_name = f'{log_file_name}.json.gz'
+                            stdout.write(4 * ' ' + f'generating backup file: {log_file_name}', ending=' ')
+                            with storage.open(log_file_name, 'wb') as f:
+                                with TextIOWrapper(gzip.GzipFile(filename='{}.json'.format(log_file_name),
+                                                                 fileobj=f, mode='wb')) as gzf:
+                                    created_log_files.append(log_file_name)
+                                    json.dump(
+                                        get_lazy_serialize_iterable(
+                                            batch_data, stdout, settings.ELASTICSERACH_CLEAN_LOGS_READ_BATCH_SIZE
+                                        ),
+                                        gzf, cls=DjangoJSONEncoder, indent=5
+                                    )
+                            stdout.write('')
+                    except Exception:
```

Comment:
exception should remove created files firstly because next run will create files again